common/lib/dependabot/shared_helpers.rb

```diff
@@ -25,14 +26,19 @@ module SharedHelpers
 
     def self.in_a_temporary_repo_directory(directory = "/",
                                            repo_contents_path = nil,
+                                           **options,
                                            &block)
       if repo_contents_path
-        path = Pathname.new(File.join(repo_contents_path, directory)).
-               expand_path
+        path = Pathname.new(File.join(repo_contents_path, directory)).expand_path
         reset_git_repo(repo_contents_path)
         # Handle missing directories by creating an empty one and relying on the
         # file fetcher to raise a DependencyFileNotFound error
         FileUtils.mkdir_p(path)
+
+        if (workspace = Dependabot::Workspace.active_workspace)
+          return workspace.change(options[:memo], &block)
```

Comment:
As above, I think we can just pass the block for now and start setting a memo in the `FileUpdater` once we've verified our experiment and wired it up into metrics.